Jenkinsfile

```diff
@@ -342,6 +355,40 @@ pipeline {
             }
 
         }
+        stage('Math functions expressions test') {
+            when {
+                expression {
+                    !skipExpressionTests
+                }
+            }
+            agent any
+            steps {
+
+                sh """
+                    git clone --recursive https://github.com/stan-dev/stan.git
+                    cd stan && make math-revert && make clean-all && git clean -xffd && git submodule update --init --recursive
+                """
+
+                writeFile(file: "make/local", text: "CXX=${env.CXX} -Werror ")
+
+                unstash 'ubuntu-exe'
+                // Where to copy the binary ?
+
+                script {
+                    dir("stan/lib/stan_math/") {
+                        sh "echo O=0 > make/local"
+                        withEnv(['PATH+TBB=./lib/tbb']) {
+                            try { sh "./runTests.py -j${env.PARALLEL} test/expressions" }
+                            finally { junit 'test/**/*.xml' }
+                        }
+                        withEnv(['PATH+TBB=./lib/tbb']) {
+                            sh "python ./test/expressions/test_expression_testing_framework.py"
```

Comment:
This was not explicitly stated in the issue but this second test is not required. This is a test of the "expression test framework" so this doesnt have to be run here as well.